### ISSUE TITLE
When a string and uri are mixed, it decomposes the uri into the hash object.

### ISSUE THREAD

**Issue by straleyb** [OregonDigital/ControlledVocabularyManager#480]

**Comment by lsat12357** [comment]
String value with a uri in it (uri is meant to be a string)

**Comment by carakey** [comment]
Example http://opaquenamespace.org/ns/llid

**Comment by lsat12357** [comment]
So I tried entering in three different string fields:
yadda http://blah.uoregon.edu
http://blah.uoregon.edu yadda
http://blah.uoregon.edu
the first two items appear in both git and blazegraph as strings; the third initially appears in git as a uri, but is ignored by blazegraph; after another cycle of editing and review, the unintended uri is still absent in blazegraph and has been removed from git.

**Comment by straleyb** [comment]
So this ticket states that. If a field contains a value that is a string, and a new value is added (Clicking the `+add` button) and a URI is put in as that new value, then it decomposes the uri into its subcomponents.

**Comment by straleyb** [comment]
This happens for every field with a language, and SubPropertyOf

**Comment by straleyb** [comment]
<img width="1310" alt="Screen Shot 2019-07-08 at 12 18 21 PM" src="https://user-images.githubusercontent.com/6424683/60836449-8d498b00-a17a-11e9-85ba-8dff3e907539.png">

**Comment by straleyb** [comment]
Here is what it looks like before review

<img width="1009" alt="Screen Shot 2019-07-08 at 12 29 22 PM" src="https://user-images.githubusercontent.com/6424683/60837050-14e3c980-a17c-11e9-9f50-a87d30bdbbfa.png">

**Comment by straleyb** [comment]
Here is after review

<img width="1168" alt="Screen Shot 2019-07-08 at 12 30 17 PM" src="https://user-images.githubusercontent.com/6424683/60837106-3e045a00-a17c-11e9-9667-d1c0481be6a4.png">

**Comment by straleyb** [comment]
Symptoms are as such. 

SubPropertyOf copies from Same As as well as retains its own value, it also picks up the value of the uri. 

The field in which gets the URI gets the uri's Vocabulary name from the end of the uri, it also saves the uri as a string, AND saves the URI as a uri. 

When this piece is edited, it displays as such:

<img width="1371" alt="Screen Shot 2019-07-08 at 12 33 49 PM" src="https://user-images.githubusercontent.com/6424683/60837312-b703b180-a17c-11e9-9acb-9a874ff1ff6a.png">

**Comment by straleyb** [comment]
As @lsat12357 said previously, if you delete all values except the final uri in the alternate names, the value doesnt get saved. Conversely, if a uri is left in the values of a field and saved, that is when you get the decomposed URI. Ill be posting a gif of the entire workflow to demonstrate.

**Comment by straleyb** [comment]
![cvm_bug](https://user-images.githubusercontent.com/6424683/60837617-6f315a00-a17d-11e9-84ca-958b05feb703.gif)

**Comment by straleyb** [comment]
Here is how to get the weird looking hash in Rails console
```
[11] pry(main)> uri = RDF::URI('http://www.opaquenamespace.org/ns/Blah')
=> #<RDF::URI:0x3ff28128da44 URI:http://www.opaquenamespace.org/ns/Blah>
[12] pry(main)> puts uri.to_hash
{:scheme=>"http", :authority=>"www.opaquenamespace.org", :userinfo=>nil, :user=>nil, :password=>nil, :host=>"www.opaquenamespace.org", :port=>nil, :path=>"/ns/Blah", :query=>nil, :fragment=>nil}
=> nil
```

**Comment by straleyb** [comment]
Fixed by #548
